mlir/lib/Dialect/Linalg/Transforms/Fusion.cpp

```diff
@@ -235,6 +237,39 @@ mlir::linalg::fuseProducerOfTensor(OpBuilder &b, OpOperand &consumerOpOperand) {
   return fuseProducerOfTensor(b, producerOpResult, consumerOpOperand);
 }
 
+/// Create tensor.collapse_shape to drop dimensions in `dropDims` in tensor
+/// `from`.
```

Comment:
Since this is only used for dropping unit dims ("rank reducing extract slice"), I would make that clear and also add an assert. Generalise is fine, but then we should test more.